server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -1097,6 +1074,45 @@ public Pool find(String poolId) {
     }
 
     /**
+     * Performs the work of the entitleByProducts method in its own transaction to help unlock
+     * pools which can no longer be bound.
+     * <p></p>
+     * This method should not be called directly, and is only declared protected to allow the
+     * @Transactional annotation to function.
+     *
+     * @param data
+     *  The autobind data to use for entitling a consumer
+     *
+     * @return
+     *  a list of entitlements created as for this autobind operation
+     */
+    @Transactional
+    protected List<Entitlement> entitleByProductsImpl(AutobindData data) throws EntitlementRefusedException {
+        Consumer consumer = data.getConsumer();
+        String[] productIds = data.getProductIds();
+        Collection<String> fromPools = data.getPossiblePools();
+        Date entitleDate = data.getOnDate();
+        Owner owner = consumer.getOwner();
+
+        List<PoolQuantity> bestPools = new ArrayList<PoolQuantity>();
+        // fromPools will be empty if the dev pool was already created.
+        if (consumer != null && consumer.isDev() && !fromPools.isEmpty()) {
```

Comment:
Autobind data not having a consumer? Not sure, the block is almost entirely as it was before I moved it.